src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import core.basesyntax.enums.FigureType;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final Random RANDOM = new Random();
+    public static final double MAX_RANGE_VALUE = 100;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        int indexRandom = RANDOM.nextInt(FigureType.values().length);
+        String figureName = FigureType.values()[indexRandom].name();
+
+        return switch (figureName) {
+            case "SQUARE" -> new Square(colorSupplier.getRandomColor(),
```

Comment:
```suggestion
            case "SQUARE" -> new Square(randomColor,
```

Change in every case